decidim-core/spec/system/search_spec.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "Search", type: :system do
+  let(:organization) { create(:organization) }
+  let(:term) { "dolorem" }
+
+  before do
+    switch_to_host(organization.host)
+    visit decidim.root_path
+  end
+
+  it "has a topbar search form" do
+    expect(page).to have_selector(".topbar__search")
+  end
+
+  context "when searching from the top bar" do
+    before do
+      within ".topbar__search" do
+        fill_in "term", with: term
+        find("input#term").native.send_keys :enter
+      end
+    end
+
+    it "displays the results page" do
```

Comment:
I think we need more system specs, maybe searching proposals and meetings and more use cases

Reply:
I have done integration tests to validate that Meetings and Proposals are indexed and to validate that the search engine finds and filters them:
- https://github.com/decidim/decidim/pull/3042/files#diff-c22b41255dea3ab8ae0c7c2a9a1aa021
- https://github.com/decidim/decidim/pull/3042/files#diff-ce7ec833cd05fe7727b518c9ae5edfb9

There are also system specs for the search at core as you already haves seen: https://github.com/decidim/decidim/pull/3042/files#diff-1c07aba5e89fe934d6ad8edcd28e30cd

Adding more system specs for Meetings and Proposals would re-re-test the search don't you think?
Or you prefer to have the search related unit tests removed and leave only the indexing related unit tests, and then test the search at a resource level with system tests?

Reply:
> Or you prefer to have the search related unit tests removed and leave only the indexing related unit tests, and then test the search at a resource level with system tests?

Maybe that's a good option yes. I just think that the system specs are really thin, maybe adding a simple happy path case for the system tests would be enough (making sure we can display proposals and meetings). And keep the edge cases and other cases for the unit tests.

Reply:
Isn't this a use case for `shared_examples` maybe? Each component implementing search should validate its resources are actually shown when searching.

Reply:
We're doing the same with `follows` and `fingerprints`. Shouldn't take that much to implement.

Reply:
Good point @josepjaume